clippy_lints/src/semicolon_block.rs

```diff
@@ -0,0 +1,134 @@
+use clippy_utils::diagnostics::{multispan_sugg_with_applicability, span_lint_and_then};
+use rustc_errors::Applicability;
+use rustc_hir::{Block, Expr, ExprKind, Stmt, StmtKind};
+use rustc_lint::{LateContext, LateLintPass, LintContext};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::Span;
+
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// For () returning expressions, check that the semicolon is inside the block.
```

Comment:
Thanks! I think this is the last thing:

I believe this lint would apply to expressions returning any type rather than just `()`

Reply:
Ah, I forgot to change that good catch. Unsure what wording to take here though, let me know if this works.

Reply:
It's tricky, maybe something like "Suggests moving the semicolon from a block's final expression outside of the block" and the opposite, to me the current reading suggests it wouldn't work on things like `{ a; b; }`

Reply:
You can rebase on master also to fix the remark CI issue